// retort-js/dist/retort.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.retort = void 0;
const conversation_1 = require("./conversation");
const id_1 = require("./id");
function retort(chatFunction) {
    let retortId = (0, id_1.id)("retort");
    let run = (autoHandleStreaming = true, ...values) => {
        const conversation = new conversation_1.RetortConversation();
        conversation.settings.isStreaming = autoHandleStreaming;
        async function runInner() {
            return chatFunction(conversation, ...values);
        }
        let executing = runInner();
        let scriptInProgress = {
            retortId: retortId,
            $: conversation,
            completionPromise: executing,
        };
        return scriptInProgress;
    };
    let returnedModule = {
        retortId: retortId,
        _run: run,
        retortType: "retort",
    };
    // Only run the chat function if this module is the main module.
    setTimeout(() => {
        if (returnedModule.retortId === require.main?.exports?.scriptId) {
            returnedModule._run();
        }
    }, 0);
    return returnedModule;
}
exports.retort = retort;
